src/box/box.cc

```diff
@@ -3499,6 +3515,28 @@ bootstrap_from_master(struct replica *master)
 	/* Make the initial checkpoint */
 	if (gc_checkpoint() != 0)
 		panic("failed to create a checkpoint");
+
+	return 0;
+}
+
+/**
+ * If bootstrap failed from remote master rebootstrap it,
+ * in the case of non-critical error.
+ */
+static void
+rebootstrap_from_master(struct replica *master)
+{
+	do {
+		box_restart_replication();
+
+		master = replicaset_find_join_master();
+		assert(master == NULL || master->applier != NULL);
+
+		if (master == NULL || master->applier == NULL ||
```

Comment:
It's not an error when master == NULL.
It simply means that this node is the best bootstrap leader available.
So you should call bootstrap_master() in this case.